tests/phpunit/admin/test-class-wc-rest-stripe-locations-controller.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/**
+ * Class WC_REST_Stripe_Locations_Controller.
+ *
+ * @package WooCommerce_Stripe/Tests/WC_REST_Stripe_Locations_Controller
+ */
+
+/**
+ * WC_REST_Stripe_Locations_Controller unit tests.
+ *
+ * @runTestsInSeparateProcesses
+ * @preserveGlobalState disabled
+ */
+class WC_REST_Stripe_Locations_Controller_Test extends WP_UnitTestCase {
+
+	/**
+	 * Tested REST route.
+	 */
+	const LOCATIONS_REST_BASE = '/wc/v3/wc_stripe/terminal/locations';
+
+	public function test_create_location_missing_params_fails() {
+		wp_set_current_user( 1 );
+
+		// Missing display_name or address parameter should be a bad request.
+		$request  = new WP_REST_Request( 'POST', self::LOCATIONS_REST_BASE );
+		$response = rest_do_request( $request );
+		$this->assertEquals( 400, $response->get_status() );
+	}
+
+	public function test_create_location_requires_auth() {
+		wp_set_current_user( 0 );
+
+		// Unauthenticated user should not be able to access route.
+		$request = new WP_REST_Request( 'POST', self::LOCATIONS_REST_BASE );
+		$request->set_param( 'display_name', 'Test Store' );
+		$request->set_param(
+			'address',
+			[
+				'line1'       => '1 Example St.',
+				'city'        => 'Example City',
+				'country'     => 'US',
+				'state'       => 'CA',
+				'postal_code' => '12345',
+			]
+		);
+		$response = rest_do_request( $request );
+		$this->assertEquals( 401, $response->get_status() );
+	}
+
+	public function test_create_location_success() {
+		wp_set_current_user( 1 );
+
+		// Mock response from Stripe API using request arguments.
+		$test_request = function ( $preempt, $parsed_args, $url ) {
+			return [
+				'response' => 200,
+				'headers'  => [ 'Content-Type' => 'application/json' ],
+				'body'     => json_encode(
+					[
+						'id'           => 'tml_12345678901234567890123',
+						'object'       => 'terminal.location',
+						'display_name' => $parsed_args['body']['display_name'],
```

Comment:
Looks like this is causing an array offset error in PHPUnit tests.

Reply:
Thanks, this should be fixed now!